onnxruntime/core/framework/execution_frame.cc

```diff
@@ -256,6 +260,13 @@ Status ExecutionFrame::AllocateAsPerAllocationPlan(int mlvalue_index,
   if (mlvalue_index < 0 || mlvalue_index >= all_values_.size())
     return Status(ONNXRUNTIME, INVALID_ARGUMENT,
                   "Tried to allocated with invalid mlvalue index: " + std::to_string(mlvalue_index));
+
+  // if there is a custom allocator for this mlvalue_index, call it to do the allocation
+  auto custom_alloc_entry = custom_allocators_.find(mlvalue_index);
```

Comment:
Is the index for the custom allocator lookup required to be a mlvalue_index or the output number? (See related comment below.)

Reply:
The input to ExecutionFrame ctor matches the order of the feeds, and ExecutionFrame::Init converts that to an overall mlvalue_index so we can do this matching.

---
In reply to: [251689399](https://github.com/Microsoft/onnxruntime/pull/377#discussion_r251689399) [](ancestors = 251689399)